livekit-plugins/livekit-plugins-speechmatics/livekit/plugins/speechmatics/tts.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+import asyncio
+import os
+from dataclasses import dataclass, replace
+
+import aiohttp
+
+from livekit.agents import (
+    APIConnectionError,
+    APIConnectOptions,
+    APIStatusError,
+    APITimeoutError,
+    tokenize,
+    tts,
+    utils,
+)
+from livekit.agents.types import (
+    DEFAULT_API_CONNECT_OPTIONS,
+    NOT_GIVEN,
+    NotGivenOr,
+)
+from livekit.agents.utils import is_given
+
+from .utils import get_tts_url
+
+BASE_URL = "https://preview.tts.speechmatics.com"
+NUM_CHANNELS = 1
+DEFAULT_VOICE = "sarah"
+
+
+@dataclass
+class _TTSOptions:
+    voice: str
+    sample_rate: int
+    word_tokenizer: tokenize.WordTokenizer
+    base_url: str
+    api_key: str
+
+
+class TTS(tts.TTS):
+    def __init__(
+        self,
+        *,
+        voice: str = DEFAULT_VOICE,
+        sample_rate: int = 16000,
+        api_key: str | None = None,
+        base_url: str = BASE_URL,
+        word_tokenizer: NotGivenOr[tokenize.WordTokenizer] = NOT_GIVEN,
+        http_session: aiohttp.ClientSession | None = None,
+    ) -> None:
+        """
+        Create a new instance of Speechmatics TTS.
+
+        Args:
+            voice (str): Voice model to use for synthesis. Options: "sarah", "theo", "megan". Defaults to "sarah".
+            sample_rate (int): Sample rate of audio. Defaults to 16000.
+            api_key (str): Speechmatics API key. If not provided, will look for SPEECHMATICS_API_KEY in environment.
+            base_url (str): Base URL for Speechmatics TTS API. Defaults to "https://preview.tts.speechmatics.com"
+            word_tokenizer (tokenize.WordTokenizer): Tokenizer for processing text. Defaults to basic WordTokenizer.
+            http_session (aiohttp.ClientSession): Optional aiohttp session to use for requests.
+        """
+        super().__init__(
+            capabilities=tts.TTSCapabilities(streaming=False),
+            sample_rate=sample_rate,
+            num_channels=NUM_CHANNELS,
+        )
+
+        api_key = api_key or os.environ.get("SPEECHMATICS_API_KEY")
+        if not api_key:
+            raise ValueError(
+                "Speechmatics API key required. Set SPEECHMATICS_API_KEY or provide api_key."
+            )
+
+        if not is_given(word_tokenizer):
+            word_tokenizer = tokenize.basic.WordTokenizer(ignore_punctuation=False)
+
+        self._opts = _TTSOptions(
+            voice=voice,
+            sample_rate=sample_rate,
+            word_tokenizer=word_tokenizer,
+            base_url=base_url,
+            api_key=api_key,
+        )
+        self._session = http_session
+
+    def _ensure_session(self) -> aiohttp.ClientSession:
+        if not self._session:
+            self._session = utils.http_context.http_session()
+        return self._session
+
```

Comment:
adding the model and provider properties

```suggestion
    @property
    def model(self) -> str:
        return "model name"

    @property
    def provider(self) -> str:
        return "Speechmatics"

```